packages/seacas/libraries/ioss/src/Ioss_DatabaseIO.C

```diff
@@ -1362,17 +1361,25 @@ namespace Ioss {
     return {xx.first, yy.first, zz.first, xx.second, yy.second, zz.second};
   }
 
-  std::vector<size_t> DatabaseIO::get_all_block_field_data(const std::string &field_name,
-                                                           void *data, size_t data_size) const
+#ifndef DOXYGEN_SKIP_THIS
+  template
+  std::vector<size_t> DatabaseIO::get_entity_field_data_internal(const std::string &field_name,
+                                                                 const std::vector<Ioss::ElementBlock*>& elem_blocks,
+                                                                 void *data, size_t data_size) const;
+#endif
+
+  template <typename T>
+  std::vector<size_t> DatabaseIO::get_entity_field_data_internal(const std::string &field_name,
+                                                                 const std::vector<T*>& entity_container,
+                                                                 void *data, size_t data_size) const
   {
-    const Ioss::ElementBlockContainer &elem_blocks = get_region()->get_element_blocks();
-    size_t                             num_blocks  = elem_blocks.size();
+    size_t num_container = entity_container.size();
 
-    std::vector<size_t> offset = get_all_block_offsets(field_name, elem_blocks);
+    std::vector<size_t> offset = get_entity_offsets(field_name, entity_container);
 
-    for (size_t i = 0; i < num_blocks; i++) {
+    for(size_t i=0; i<num_container; i++) {
 
-      Ioss::ElementBlock *entity = elem_blocks[i];
+      auto entity = entity_container[i];
```

Comment:
This should probably be `const auto *entity = entity_container[i];`

Reply:
I can change that too if you want